src/Controller/AdminController.php

```diff
@@ -0,0 +1,249 @@
+            $this->logger->error($message);
+            return new JsonResponse($message, 400);
+        } else {
+            $amountInCents = $this->currencyUtil->sanitize($order->getAmountTotal(), null);
+            if (!$this->refundService->isAmountRefundable($order, $amountInCents)) {
+                return new JsonResponse([
+                    'success' => false,
+                    'message' => 'adyen.invalidRefundAmount',
+                ]);
+            }
+        }
+
+        try {
+            $result = $this->refundService->refund($order);
+            // If response does not contain pspReference
+            if (!array_key_exists('pspReference', $result)) {
+                $message = sprintf('Invalid response for refund on order %s', $order->getOrderNumber());
+                throw new AdyenException($message);
+            }
+
+            $this->refundService->insertAdyenRefund(
+                $order,
+                $result['pspReference'],
+                RefundEntity::SOURCE_SHOPWARE,
+                RefundEntity::STATUS_PENDING_WEBHOOK
+            );
+        } catch (\Exception $e) {
+            $this->logger->error($e->getMessage());
+
+            return new JsonResponse([
+                'success' => false,
+                'message' => 'adyen.refundError',
+            ]);
+        }
+
+        return new JsonResponse(['success' => true]);
+    }
+
+    /**
+     * @Route(
+     *     "/api/v{version}/adyen/orders/{orderId}/refunds",
+     *     name="api.adyen_refund.get",
+     *     methods={"GET"}
+     * )
+     *
+     * @param string $orderId
+     * @return JsonResponse
+     */
+    public function getRefunds(string $orderId): JsonResponse
+    {
+        $refunds = $this->adyenRefundRepository->getRefundsByOrderId($orderId);
+
+        return new JsonResponse($this->buildRefundResponseData($refunds->getElements()));
+    }
+
+    /**
+     * Build a response containing the data related to the refunds
+     *
+     * @param array $refunds
+     * @return array
+     */
+    private function buildRefundResponseData(array $refunds)
+    {
+        $context = Context::createDefaultContext();
+        $result = [];
+        /** @var RefundEntity $refund */
+        foreach ($refunds as $refund) {
+            $updatedAt = $refund->getUpdatedAt();
+            $order = $refund->getOrderTransaction()->getOrder();
+            $amount = $this->currencyFormatter->formatCurrencyByLanguage(
+                $refund->getAmount() / 100,
+                $order->getCurrency()->getIsoCode(),
+                $order->getLanguageId(),
+                $context
+            );
+            $result[] = [
+                'pspReference' => $refund->getPspReference(),
+                'amount' => $amount,
+                'rawAmount' => $refund->getAmount(),
+                'status' => $refund->getStatus(),
+                'createdAt' => $refund->getCreatedAt()->format('Y-m-d H:i (e)'),
```

Comment:
I think it would be a nice addition to use the constant created in the other PR for the date format.